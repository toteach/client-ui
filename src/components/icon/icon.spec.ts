import { mount, Wrapper } from '@vue/test-utils';
import Vue from 'vue';

import TIcon from './icon.vue';

describe('TIcon', () => {
  let wrapper: Wrapper<Vue>;
  const iconName = 'test';
  beforeEach(() => {
    wrapper = mount(TIcon, {
      propsData: {
        name: iconName,
      },
    });
  });
  it('is initialized', () => {
    expect(wrapper.name()).toBe('TIcon');
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('should contain default svg', () => {
    expect(wrapper.contains('svg')).toBeTruthy();
  });
  it('should set correct name classes', () => {
    expect(wrapper.is('span')).toBeTruthy();
    expect(wrapper.classes('t-icon')).toBeTruthy();
    expect(wrapper.classes(`t-icon_name_${iconName}`)).toBeTruthy();
  });
});
