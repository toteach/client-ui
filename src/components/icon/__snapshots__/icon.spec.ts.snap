// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TIcon renders correctly 1`] = `"<span class=\\"t-icon t-icon_name_test\\"><svg class=\\"t-icon__svg\\"><use xlink:href=\\"#t-icon-test\\"></use></svg></span>"`;
